cluster/terraform/tls.tf

```diff
@@ -0,0 +1,31 @@
+data "azurerm_key_vault" "cert_kv" {
+  name                = var.cluster_kv
+  resource_group_name = var.resource_group_name
+}
+
+data "azurerm_key_vault_certificate_data" "cert" {
+  name         = local.cluster_cert_secret
+  key_vault_id = data.azurerm_key_vault.cert_kv.id
+}
+
+locals {
+  # The cert from the KV contains the full chain and is in the wrong order
+  # So we need to extract the primary cert which is the last in the list
```

Comment:
Can you reference https://github.com/kubernetes/ingress-nginx/blob/1930b1467d93238bbcd96daac8e3863a91e58747/docs/user-guide/tls.md#tls-secrets in the comment?

Reply:
added